Introduce sb_is_modular build flag

niranjanyardi opened the pull request:
b/246854012

sb_is_modular is set for modular builds, which have:
1) Application binary built as a shared library.
2) Either of a) or b)
   a. Starboard built at a shared library and a separate loader_app executable.
   b. A loader_app executable with Starboard built in ( Evergreen ).

sb_is_modular shares a large subset with evergreen but is different
because of certain elements which are evergreen specific. An example of
evergreen specific code not part of modular builds is
a)Cobalt app updater(cobalt/updater/updater.cc)

This change replaces certain common instances of sb_is_evergreen with sb_is_modular
and SB_IS_EVERGREEN with SB_IS_MODULAR

kaidokert commented:
Quick look, this seems like a sensible simplification to me. Worth calling out somewhere that SB_IS_EVERGREEN implies MODULAR from now on as well.

y4vor commented on cobalt/css_parser/scanner_test.cc:
```diff
@@ -42,7 +42,8 @@ class ScannerTest : public ::testing::Test,
     }
 
-#if SB_IS(EVERGREEN)
+// TODO: Ask kaido/yavor if this is okay
```
Yeah, this is OK to check for modular.

y4vor commented on starboard/client_porting/icu_init/icu_init.cc:
```diff
@@ -44,7 +44,8 @@ void Initialize() {
   data_path += "icu";
 
-#if SB_IS(EVERGREEN)
+// TODO: Ask Kaido/YAvor if this is needed.
```
Let's leave this one to be Evergreen. Also we should actually clean it up as we changed where we store the ICU tables. But that would be different change.

y4vor commented on third_party/inspector_protocol/crdtp/json_platform.cc:
```diff
@@ -16,7 +16,8 @@ namespace crdtp {
 namespace platform {
 bool StrToD(const char* str, double* result) {
-#if SB_IS(EVERGREEN)
+//Ask Kaido/Yavor if this is needed.
```
Yes, this should be modular.

y4vor commented on third_party/inspector_protocol/crdtp/json_platform.cc:
```diff
@@ -26,7 +27,8 @@ bool StrToD(const char* str, double* result) {
 
 std::string DToStr(double value) {
-#if SB_IS(EVERGREEN)
+  //Ask Kaido/Yavor if this is needed.
```
Yes, this should be modular.

niranjanyardi replied:
done

y4vor commented on starboard/BUILD.gn:
```diff
@@ -57,7 +57,7 @@ group("gn_all") {
   }
 
-  if (!sb_is_evergreen) {
+  if (!sb_is_modular) {
```
This one should stay Evergreen. We use LZ4 for the updates only.

y4vor commented on cobalt/BUILD.gn:
```diff
@@ -66,22 +66,26 @@ group("default") {
-      "//components/crx_file",
-      "//components/prefs",
-      "//components/update_client",
       "//third_party/llvm-project/compiler-rt:compiler_rt",
```
We probably don't need those dependencies here at all as add them whenever you depend on the starboard platform. You can keep them for now, but we should try to clean it up in separate change.

niranjanyardi commented on starboard/export.h:
```diff
@@ -24,12 +24,12 @@
 // with all client applications being built separately.
 
-#if defined(COMPONENT_BUILD) || SB_IS(EVERGREEN)
+#if defined(COMPONENT_BUILD) || SB_IS(MODULAR)
```
talked to yavor, not needed

niranjanyardi commented:
> Quick look, this seems like a sensible simplification to me. Worth calling out somewhere that SB_IS_EVERGREEN implies MODULAR from now on as well.

I added check in starboard/BUILD.gn asserting when sb_is_modular should be set.
Also changed the commit description

kaidokert commented:
Lets add a more descriptive thing here into PR description explaining where this flag came from, how does it differ from sb_is_evergreen and the extent of intended use. Also recommend renaming the title to "Introducing sb_is_modular build flag" to be really clean and explicit about when this was brought in as a concept.

kaidokert reviewed (approved):
Otherwise LGTM - but please make the description more extensive. Think of a developer reading this commit message 4 years hence.

niranjanyardi commented:
> Otherwise LGTM - but please make the description more extensive. Think of a developer reading this commit message 4 years hence.

changed title and first comment in the "conversations" view
